### PR TITLE
Improve 'dsp' test to check that checksum option is set correctly.

### PR DESCRIPTION
**Pull request by hlinnaka**
The 'dsp' test tests that the gp_default_storage_options GUC affects
created tables correctly. Some of the tables were created with checksums
on, some off. However, the catalog queries used to verify that the tables
were indeed created with the correct options didn't include the 'checksum'
column, so even though the option was set, it was not validated like the
other table options like compression type. Add 'checksum' to the validation
queries.

Remove bespoken TINC test that was testing for that. Now that 'dsp' tests
for that, it seems redundant.

### CODE REVIEW

File: src/test/regress/sql/dsp.sql [hunk 1]
```diff
@@ -132,7 +132,7 @@ create table co7(
 	b varchar encoding(compresstype=zlib,compresslevel=6),
 	c char encoding(compresstype=rle_type))
 	with (checksum=false) distributed by (a);
-select relid::regclass,compresslevel,compresstype,blocksize,checksum,columnstore
+select relid::regclass,compresslevel,compresstype,blocksize,checksum,columnstore,checksum
```
Comment (asubramanya) [inline src/test/regress/sql/dsp.sql]: Why select `checksum` twice?

File: src/test/regress/sql/dsp.sql [hunk 2]
```diff
@@ -164,7 +164,7 @@ create table co4 (a int encoding (blocksize=8192), b int) distributed by (a);
 set gp_default_storage_options='appendonly=true';
 create table co5 (a int encoding (blocksize=8192), b float)
 	with (orientation=column, checksum=false) distributed by (a);
-select relid::regclass,compresslevel,compresstype,blocksize,checksum,columnstore
+select relid::regclass,compresslevel,compresstype,blocksize,checksum,columnstore,checksum
```
Comment (asubramanya) [inline src/test/regress/sql/dsp.sql]: Same here.

File: src/test/regress/sql/dsp.sql (hunk 1 above)
Comment (hlinnaka) [inline src/test/regress/sql/dsp.sql]: Oh, I didn't notice that it's there already. I skimmed over too quickly and thought all the queries were the same. That means that the test actually covered pretty well what the TINC test did.

Removed it from the queries that already had it, and updated the commit message accordingly. Please have a second look.

**Comment by hlinnaka** [comment]
Ok, pushed. Thanks for the review!

**Pull request closed without merging**